signer/tests/integration/transaction_coordinator.rs

```diff
@@ -3422,6 +3426,428 @@ async fn test_conservative_initial_sbtc_limits() {
+    // -------------------------------------------------------------------------
+    // - Before the signers can process anything, they need a UTXO to call
+    //   their own. For that we make a donation, and confirm it. The
+    //   signers should pick it up.
+    // =========================================================================
+    let script_pub_key = shares.aggregate_key.signers_script_pubkey();
+    let network = bitcoin::Network::Regtest;
+    let address = Address::from_script(&script_pub_key, network).unwrap();
+
+    faucet.send_to(100_000_000, &address);
+
+    // =========================================================================
+    // Step 6 - Receive the withdrawal request
+    // -------------------------------------------------------------------------
+    // - Withdrawal requests are received from our stacks node, where we
+    //   are an event observer of our node. So creating a withdrawal
+    //   request is basically writing a row to our database.
+    // =========================================================================
+    let withdrawal_recipient = Recipient::new(AddressType::P2tr);
+
+    // Let's initiate a withdrawal. This is easy, we just create a row in
+    // the databases.
+    let (bitcoin_chain_tip, stacks_chain_tip) = db.get_chain_tips().await;
+
+    let withdrawal_request = WithdrawalRequest {
+        request_id: 23,
+        bitcoin_block_height: bitcoin_chain_tip.block_height,
+        amount: 10_000_000,
+        block_hash: stacks_chain_tip,
+        recipient: withdrawal_recipient.script_pubkey.clone().into(),
+        max_fee: 100_000,
+        txid: StacksTxId::from([123; 32]),
+        sender_address: PrincipalData::from(StandardPrincipalData::transient()).into(),
+    };
+    for (_, db, _, _) in signers.iter() {
+        db.write_withdrawal_request(&withdrawal_request)
+            .await
+            .unwrap();
+    }
+
+    // =========================================================================
+    // Step 7 - Confirm WITHDRAWAL_MIN_CONFIRMATIONS more blocks so that
+    //          the signers process the withdrawal request.
+    // -------------------------------------------------------------------------
+    // - Generate WITHDRAWAL_MIN_CONFIRMATIONS blocks the deposit request.
+    //   This will trigger the block observer to update the database.
+    // - Each RequestDecider process should vote on the withdrawal request
+    //   and submit the votes to each other.
+    // - The coordinator should submit a sweep transaction with an output
+    //   fulfilling the withdrawal request.
+    // =========================================================================
+    for _ in 0..WITHDRAWAL_MIN_CONFIRMATIONS {
+        faucet.generate_blocks(1);
+        wait_for_signers(&signers).await;
+    }
+
+    let (ctx, _, _, _) = signers.first().unwrap();
+    let mut txids = ctx.bitcoin_client.inner_client().get_raw_mempool().unwrap();
```

Comment:
Maybe we should do this after each `wait_for_signers()` in the loop above as well, but assert 0 (until `min_confs - 1` i think)? Just since we're anyways like 350 lines into the test and can pretty easily check that the min confirmations limit is being respected?

Reply:
Good point, updated in https://github.com/stacks-network/sbtc/pull/1403/commits/3105605f75746d0eaba7a86215daaf8142683f4d